pkg/kubectl/cmd/create_pvc.go

```diff
@@ -0,0 +1,222 @@
+)
+
+type CreatePVCOptions struct {
+	Name         string
+	AccessModes  []api.PersistentVolumeAccessMode
+	Capacity     *resource.Quantity
+	StorageClass *string
+	VolumeName   string
+
+	DryRun       bool
+	OutputFormat string
+	Namespace    string
+	Client       internalversionstorage.PersistentVolumeClaimInterface
+	Mapper       meta.RESTMapper
+	Out          io.Writer
+	PrintObject  func(obj runtime.Object) error
+}
+
+// PersistentVolumeClaim is a command to ease creating PersistentVolumeClaims.
+func NewCmdCreatePersistentVolumeClaim(f cmdutil.Factory, cmdOut io.Writer) *cobra.Command {
+	c := &CreatePVCOptions{
+		Out: cmdOut,
+	}
+	cmd := &cobra.Command{
+		Use:     "persistentvolumeclaim NAME --access-mode=accessmode --capacity=capacity [--storage-class=storageclass] [--volume-name=volumename] [--dry-run]",
+		Short:   pvcLong,
+		Long:    pvcLong,
+		Example: pvcExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(c.Complete(f, cmd, args))
+			cmdutil.CheckErr(c.Validate())
+			cmdutil.CheckErr(c.RunCreatePVC())
+		},
+	}
+	cmdutil.AddApplyAnnotationFlags(cmd)
+	cmdutil.AddValidateFlags(cmd)
+	cmdutil.AddPrinterFlags(cmd)
+	cmdutil.AddDryRunFlag(cmd)
+
+	cmd.Flags().StringSlice("access-mode", []string{}, "AccessMode contains the desired access modes the volume should have.")
+	cmd.Flags().String("capacity", "", "Capacity represents the minimum storage capacity required.")
+	cmd.Flags().String("storage-class", "", "Name of the StorageClass required by the claim.")
+	cmd.Flags().StringVar(&c.VolumeName, "volume-name", "", "VolumeName is the binding reference to the PersistentVolume backing this claim.")
+
+	return cmd
+}
+
+func (c *CreatePVCOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
+	name, err := NameFromCommandArgs(cmd, args)
+	if err != nil {
+		return err
+	}
+	c.Name = name
+
+	// Transform access mode abbreviation to complete access mode and remove duplicate access mode.
+	accessModes := []api.PersistentVolumeAccessMode{}
+	modes := cmdutil.GetFlagStringSlice(cmd, "access-mode")
+	for _, m := range modes {
+		var accessMode api.PersistentVolumeAccessMode
+		switch m {
+		case "RWX", string(api.ReadWriteMany):
+			accessMode = api.ReadWriteMany
+		case "RWO", string(api.ReadWriteOnce):
+			accessMode = api.ReadWriteOnce
+		case "ROX", string(api.ReadOnlyMany):
+			accessMode = api.ReadOnlyMany
+		default:
+			return fmt.Errorf("invalid access mode: %s", m)
+		}
+
+		containsMode := false
+		for _, a := range accessModes {
+			if a == accessMode {
+				containsMode = true
+				break
+			}
+		}
+
+		if !containsMode {
+			accessModes = append(accessModes, accessMode)
+		}
+	}
+	c.AccessModes = accessModes
+
+	// If user does not provide storage-class option, then StorageClass will be set as nil.
+	// Otherwise, StorageClass will be the value set by user, including empty string.
+	storageClassFlag := cmd.Flags().Lookup("storage-class")
```

Comment:
This flag has been set in L92. No need to double check its existences. We can add checking in tests.

Reply:
we need to distinguish some different cases here:
1. User does not specify storage-class in the command line, so the generated PVC will only use the default storage class. e.g. standard, this is why I added a line here to detect whether this flag is specified by user. L92 is an default value set, but I don't know whether there is a better way to detect whether this flag is set by user or not.
2. User specify an empty storage-class, like `--storage-class=` or `--storage-class=""`, in this case, the storage class in generated PVC shall also be empty string.
3. User specify an normal storage-class like `--storage-class=abc`, the generated PVC's storage-class should be `abc` as well.

Reply:
Actually I mean `storageClassFlag != nil` can be omitted. We can test this flag in our test cases.

Reply:
ok, I see, I can remove the check then

Reply:
@xingzhou I think there is no need to make such distinction.

https://github.com/kubernetes/kubernetes/blob/5f184c56b37b3b25f8d9d30bc6117c0ac8eb6b34/pkg/api/helper/helpers.go#L593-L606

Since we have already set a default value `""` to `StorageClass`, no matter the value is explicitly set 
 ( like `--storage-class=abc`) or not, nothing will get affected. We only need to get this value by `storageClass := cmdutil.GetFlagString(cmd, "storage-class")`. Or we use `cmd.Flags().StringP("storage-class",xxxxx)` in L92.

Reply:
not sure if I understand correctly, in your suggestion, finally there would be only two possibilities:
1. the PVC's storage class will be set as empty string(the user may not specify the option, or provide an empty string as an option)
2. the PVC's storage class is a common string(like "abc")
but I think we need to provide a way for user to choose use default storage class, which means that the `PersistentVolumeClaim`'s `storageClassName` field( *string ) is `nil`

Reply:
@gnufied, when we are creating PVC's, for the following 3 different cases:
1. storage-class is ""
2. storage-class not provided(nil)
3. storage-class is a common name(like "abc")
I think this will affect different following behavior of how the following dynamic provisioning is performed?

Reply:
L92 has already assigned a default empty string, right?

So `storageClass := cmdutil.GetFlagString(cmd, "storage-class")` will be an empty string or other string value, right? `c.StorageClass = &storageClass` will not be a nil pointer.

Reply:
yes, but I have a check here:
```
storageClassFlag != nil && storageClassFlag.Changed
```
I also had some tests to test whether the user specified this option from cmd line, hope this works for the `nil` case